### PR TITLE
Update measurement clusters

### PR DESCRIPTION
**Pull request by TedTolboom**
Major completion of the measurement clusters (to enable further device integration).

### CODE REVIEW

File: lib/clusters/analogInput.js [hunk 1]
```diff
@@ -1,8 +1,31 @@
...
+    noFaultDetected: 0,     // NO-FAULT-DETECTED (0)
+    noSensor: 1,            // NO-SENSOR (1) - for input clusters only
+    overRange: 2,           // OVER-RANGE (2)
+    underRange: 3,          // UNDER-RANGE (3)
+    openLoop: 4,            // OPEN-LOOP (4)
+    shortedLoop: 5,         // SHORTED-LOOP (5)
+    noOutput: 6,            // NO-OUTPUT (6) - for input clusters only
+    unreliableOther: 7,     // UNRELIABLE-OTHER (7)
+    processError: 8,        // PROCESS-ERROR (8)
+    // multiStateFault: 9   // MULTI-STATE-FAULT (9) - for multistate clusters only
```
Comment (RobinBol) [inline lib/clusters/analogInput.js:22]: This one should be removed right?

File: lib/clusters/analogOutput.js [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
 const Cluster = require('../Cluster');
+const { ZCLDataTypes } = require('../zclTypes');
 
 const ATTRIBUTES = {
+  description: { id: 28, type: ZCLDataTypes.string },
+  maxPresentValue: { id: 65, type: ZCLDataTypes.single },
+  minPresentValue: { id: 69, type: ZCLDataTypes.single },
+  outOfService: { id: 81, type: ZCLDataTypes.bool },
+  presentValue: { id: 85, type: ZCLDataTypes.single },
+  reliability: { id: 103, type: ZCLDataTypes.enum8({
```
Comment (RobinBol) [inline lib/clusters/analogOutput.js:12]: The commented out values should be removed right?

File: lib/clusters/analogOutput.js [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+    openLoop: 4,            // OPEN-LOOP (4)
+    shortedLoop: 5,         // SHORTED-LOOP (5)
+    // noOutput: 6,            // NO-OUTPUT (6) - for input clusters only
+    unreliableOther: 7,     // UNRELIABLE-OTHER (7)
+    processError: 8,        // PROCESS-ERROR (8)
+    // multiStateFault: 9,  // MULTI-STATE-FAULT (9) - for multistate clusters only
+    configurationError: 10  // CONFIGURATION-ERROR (10)
+    })
+  },
+  relinquishDefault: { id: 104, type: ZCLDataTypes.bool },
```
Comment (RobinBol) [inline lib/clusters/analogOutput.js:26]: Shouldn't this be of type `single`?

File: lib/clusters/analogValue.js [hunk 4]
```diff
@@ -1,8 +1,29 @@
...
+    underRange: 3,          // UNDER-RANGE (3)
+    openLoop: 4,            // OPEN-LOOP (4)
+    shortedLoop: 5,         // SHORTED-LOOP (5)
+    // noOutput: 6,            // NO-OUTPUT (6) - for input clusters only
+    unreliableOther: 7,     // UNRELIABLE-OTHER (7)
+    processError: 8,        // PROCESS-ERROR (8)
+    // multiStateFault: 9,   // MULTI-STATE-FAULT (9) - for multistate clusters only
+    configurationError: 10  // CONFIGURATION-ERROR (10)
+    })
+  },
```
Comment (RobinBol) [inline lib/clusters/analogValue.js:23]: Same as above, commented values, either they are in or out :)

Comment (RobinBol) [reply lib/clusters/analogValue.js:23]: Will not repeat myself, but also check the other clusters

File: lib/clusters/binaryOutput.js [hunk 5]
```diff
@@ -1,8 +1,38 @@
...
+    openLoop: 4,            // OPEN-LOOP (4)
+    shortedLoop: 5,         // SHORTED-LOOP (5)
+    // noOutput: 6,            // NO-OUTPUT (6) - for input clusters only
+    unreliableOther: 7,     // UNRELIABLE-OTHER (7)
+    processError: 8,        // PROCESS-ERROR (8)
+    // multiStateFault: 9,   // MULTI-STATE-FAULT (9) - for multistate clusters only
+    configurationError: 10  // CONFIGURATION-ERROR (10)
+    })
+  },
+  relinquishDefault: { id: 104, type: ZCLDataTypes.bool },
```
Comment (RobinBol) [inline lib/clusters/binaryOutput.js:33]: Also here, should this be bool or single? (also check other clusters where this is used)

File: lib/clusters/deviceTemperature.js [hunk 6]
```diff
@@ -2,7 +2,17 @@
...
+const ATTRIBUTES = {
+  currentTemperature: { id: 0, type: ZCLDataTypes.int16 },
+  minTempExperienced: { id: 1, type: ZCLDataTypes.int16 },
+  maxTempExperienced: { id: 2, type: ZCLDataTypes.int16 },
+  overTempTotalDwell: { id: 3, type: ZCLDataTypes.uint16 },
+  DeviceTempAlarmMask: { id: 16, type: ZCLDataTypes.map8( 'deviceTemperatureTooLow', 'deviceTemperatureTooHigh') },
+  LowTempThreshold: { id: 3, type: ZCLDataTypes.int16 },
+  HighTempThreshold: { id: 3, type: ZCLDataTypes.int16 },
+  LowTempDwellTripPoint: { id: 3, type: ZCLDataTypes.uint24 },
+  HighTempDwellTripPoint: { id: 3, type: ZCLDataTypes.uint24 },
```
Comment (RobinBol) [inline lib/clusters/deviceTemperature.js:14]: Lets keep the casing consistent. By the way, eslint would have notified you of this!

Comment (RobinBol) [inline lib/clusters/deviceTemperature.js:14]: These IDs are not right

File: lib/clusters/illuminanceMeasurement.js [hunk 7]
```diff
@@ -5,6 +5,15 @@ const { ZCLDataTypes } = require('../zclTypes');
 
 const ATTRIBUTES = {
   measuredValue: { id: 0, type: ZCLDataTypes.uint16 },
+  minMeasuredValue: { id: 1, type: ZCLDataTypes.uint16 },
+  maxMeasuredValue: { id: 2, type: ZCLDataTypes.uint16 },
+  tolerance: { id: 3, type: ZCLDataTypes.uint16 },
+  lightSensorType: { id: 3, type: ZCLDataTypes.enum8({
```
Comment (RobinBol) [inline lib/clusters/illuminanceMeasurement.js:11]: Wrong id

File: lib/clusters/analogOutput.js (hunk 3 above)
Comment (TedTolboom) [reply lib/clusters/analogOutput.js:26]: Ah, Analog is `single` and Binary is `bool`... will check and update

File: lib/clusters/binaryOutput.js (hunk 5 above)
Comment (TedTolboom) [reply lib/clusters/binaryOutput.js:33]: No, for Binary it is `bool`: see 3.14.6.4.2

**Comment by TedTolboom** [comment]
Issues resolved and changed implemented as requested in above commits

**Pull request merged**